src/engine/sparqlExpressions/StringExpressions.cpp

```diff
@@ -380,8 +379,8 @@ using MergeRegexPatternAndFlagsExpression =
 };
 
 using ReplaceExpression =
-    StringExpressionImpl<3, decltype(replaceImpl), RegexValueGetter,
-                         ReplacementStringGetter>;
+    LiteralExpressionImpl<3, decltype(replaceImpl), RegexValueGetter,
+                          ReplacementStringGetter>;
```

Comment:
Also extend the unit tests for `replace` in `SparqlExpressionTest.cpp` to also test  inputs with dataypes and strings etc. 
(Exactly the same as you have done for `substr` in a previous PR.)